src/super_gradients/training/models/segmentation_models/unet/unet.py

```diff
@@ -0,0 +1,229 @@
+import torch.nn as nn
+from typing import Optional, Union, List
+import math
+
+from super_gradients.training.utils import HpmStruct, get_param
+from super_gradients.training import models
+from super_gradients.training.models.segmentation_models.segmentation_module import SegmentationModule
+from super_gradients.training.utils.module_utils import make_upsample_module, UpsampleMode
+from super_gradients.training.utils.module_utils import ConvBNReLU, fuse_repvgg_blocks_residual_branches
+from super_gradients.training.models.segmentation_models.unet.unet_encoder import UNetBackboneBase, Encoder
+from super_gradients.training.models.segmentation_models.context_modules import build_context_module
+from super_gradients.training.models.segmentation_models.unet.unet_decoder import Decoder
+
+
+class SegmentationHead(nn.Module):
+    def __init__(self, in_channels: int, mid_channels: int, num_classes: int, dropout: float):
+        super().__init__()
+        self.seg_head = nn.Sequential(
+            ConvBNReLU(in_channels, mid_channels, kernel_size=3, padding=1, stride=1, bias=False),
+            nn.Dropout(dropout),
+            nn.Conv2d(mid_channels, num_classes, kernel_size=1, bias=False)
+        )
+
+    def forward(self, x):
+        return self.seg_head(x)
+
+
+class UNetBase(SegmentationModule):
+    def __init__(self,
+                 num_classes: int,
+                 use_aux_heads: bool,
+                 final_upsample_factor: int,
+                 head_hidden_channels: Optional[int],
+                 head_upsample_mode: Union[UpsampleMode, str],
+                 align_corners: bool,
+                 backbone_params: dict,
+                 context_module_name: Optional[str],
```

Comment:
A little bit problematic since the `in_channels` arg is dynamic and depends on the backbone output.
I took OptimizerTypeFactory as example. Here [83ef910](https://github.com/Deci-AI/super-gradients/pull/401/commits/83ef910b7f673939f45d9bee095f0c9400228b1b)